profiling/src/collections/identifiable/mod.rs

```diff
@@ -66,13 +67,34 @@ pub trait Dedup<T: Item> {
     /// Panics if the number of items overflows the storage capabilities of
     /// the associated Id type.
     fn dedup(&mut self, item: T) -> <T as Item>::Id;
+
+    /// Deduplicate the Item, and check if the generated Id is valid.
+    fn checked_dedup(&mut self, item: T) -> anyhow::Result<<T as Item>::Id>;
```

Comment:
Having a panic in the code isn't great.  Depending on how many places dedup is used in, maybe this should be the only version and clients can `?` on it?

Reply:
I hesitated on doing that -- that `?` will need to propagate quite far. Not sure it's worth the extra work given id overflow would mean we have 2**32 items so we'd already be using many GiB's for the profiler?

Reply:
Fine for this PR, as we don't change the existing behaviour.